.github/workflows/node.yml

```diff
@@ -72,66 +65,115 @@ jobs:
+        env:
+          SECRET_ACCESS: ${{ secrets.DOCKERHUB_IMAGE_PREFIX }}
+        run: |
+          echo "Enable build and push: ${{ env.SECRET_ACCESS != '' }}"
+          echo "enable=${{ env.SECRET_ACCESS != '' }}" >> $GITHUB_OUTPUT
+
+      - name: Get the Docker tag for GHCR
+        id: ghcr-tag
+        if: steps.check-build-and-push.outputs.enable == 'true'
+        uses: docker/metadata-action@v4
+        with:
+          images: |
+            ghcr.io/${{ github.repository }}/quantel-gateway
+          tags: |
+            type=schedule
+            type=ref,event=branch
+            type=ref,event=tag
+            type=raw,value=latest,enable={{is_default_branch}}
+
+      - name: Get the Docker tag for DockerHub
+        id: dockerhub-tag
+        if: steps.check-build-and-push.outputs.enable == 'true'
+        uses: docker/metadata-action@v4
+        with:
+          images: |
+            ${{ secrets.DOCKERHUB_IMAGE_PREFIX }}
+          tags: |
+            type=schedule
+            type=ref,event=branch
+            type=ref,event=tag
+            type=raw,value=latest,enable={{is_default_branch}}
+
       - name: Build
-        if: ${{ steps.image-tags.outputs.images }}
+        if: steps.check-build-and-push.outputs.enable == 'true'
         run: |
           sudo apt-get install libomniorb4-dev
           yarn install --check-files --frozen-lockfile
           yarn build
-          yarn install --check-files --frozen-lockfile --production --force 
+          yarn install --check-files --frozen-lockfile --production --force
+
       - name: Set up Docker Buildx
-        if: ${{ steps.image-tags.outputs.images }}
-        uses: docker/setup-buildx-action@v1
+        if: steps.check-build-and-push.outputs.enable == 'true'
+        uses: docker/setup-buildx-action@v2
+
       - name: Login to DockerHub
-        if: steps.image-tags.outputs.images && steps.image-tags.outputs.dockerhub-publish == '1'
-        uses: docker/login-action@v1
+        if: steps.check-build-and-push.outputs.enable == 'true' && steps.dockerhub.outputs.dockerhub-publish == '1'
+        uses: docker/login-action@v2
         with:
           username: ${{ secrets.DOCKERHUB_USERNAME }}
           password: ${{ secrets.DOCKERHUB_TOKEN }}
-      - name: Build and push
-        uses: docker/build-push-action@v2
-        if: ${{ steps.image-tags.outputs.images }}
+
+      - name: Login to GitHub Container Registry
+        if: steps.check-build-and-push.outputs.enable == 'true' && steps.check-ghcr.outputs.enable == 'true'
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and push to GHCR
+        if: steps.check-build-and-push.outputs.enable == 'true' && steps.check-ghcr.outputs.enable == 'true'
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          file: Dockerfile.circle
```

Comment:
Quick thought, any particular reason for using the Dockerfile.circle rather than the normal Dockerfile which does the full build? (Saves doing all the yarn stuff outside of Docker)

Reply:
It's not being used, and can be removed

Reply:
I think we started with the separate files so that circleci could cache some of the node_modules between builds, but as we don't have that setup for github actions it may as well use the normal dockerfile.

I suppose having it separate does give us better error reporting back, as the error is more targetted, but I dont think we will miss that